records/state/state.go

```diff
@@ -162,7 +162,8 @@ func statusIPs(st []Status, src func(*Status) []string) []string {
 	// https://github.com/apache/mesos/blob/0.24.0/src/slave/slave.cpp#L5226-L5238
 	ts, j := -1.0, -1
 	for i := range st {
-		if st[i].State == "TASK_RUNNING" && st[i].Timestamp > ts {
+		state := st[i].State
+		if (state == "TASK_RUNNING" || state == "TASK_STAGING") && st[i].Timestamp > ts {
```

Comment:
That said, there's probably also a use case that says: "show me only the
healthy services". I think we need to support both.

On Wed, Oct 7, 2015 at 12:03 PM, Tomás Senart <notifications@github.com>
wrote:

> In records/state/state.go
> <https://github.com/mesosphere/mesos-dns/pull/308#discussion_r41409291>:
>
> > @@ -162,7 +162,8 @@ func statusIPs(st []Status, src func(*Status) []string) []string {
> >  	// https://github.com/apache/mesos/blob/0.24.0/src/slave/slave.cpp#L5226-L5238
> >  	ts, j := -1.0, -1
> >  	for i := range st {
> > -		if st[i].State == "TASK_RUNNING" && st[i].Timestamp > ts {
> > +		state := st[i].State
> > +		if (state == "TASK_RUNNING" || state == "TASK_STAGING") && st[i].Timestamp > ts {
>
> state := st[i].State[:4]if (state == "RUNNING" || state == "STAGING") && st[i].Timestamp <= ts {
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/308/files#r41409291>.
>